src/operator/random/sample_op.cc

```diff
@@ -51,7 +52,6 @@ DMLC_REGISTER_PARAMETER(SampleGenNegBinomialLikeParam);
   .set_num_outputs(1)                                                                        \
   .set_attr_parser(ParamParser<ParamType>)                                                   \
   .set_attr<nnvm::FInferShape>("FInferShape", InitShape<ParamType>)                          \
-  .set_attr<nnvm::FInferType>("FInferType", SampleOpType<ParamType>)                         \
```

Comment:
Why move this line?

Reply:
Good catch. By mistake.

Reply:
Resolved it by using template specialization as suggested by you.